### PR TITLE
6 scroll to zoom

### PR DESCRIPTION
**Pull request by paulbruzzi**
The functionality for scrolling to zoom has been added. There is a cap on the maximum distance you can zoom out. Something to note is that I used a lambda function to make this work which can probably cleaned up better.

### CODE REVIEW

**Review by mattsweet417 — CHANGES_REQUESTED** [review]
Really nice work, just a few changes that need to be added:
1. I think you should include semicolons at the end of each line (ik they arent required but are helpful for reading). I see that you have one on line 413 but nowhere else.
2. I am able to scroll while engaged inside a building view. The scroll function should stop on the map once I click on a building. Maybe even allow scrolling for a certain building? Could be useful for larger buildings.
3. Is there a way to zoom in based on where your cursor is? I see that it just zooms in on the center of the map when you zoom, regardless of where your cursor is.
4. I also notice that when you zoom in and out really fast, it seems to bug out and delete some buildings. If you can investigate this issue that'd be awesome. See:
glitchyzoom

**Review by Acumane — CHANGES_REQUESTED** [review]
- Cleared a lot of files that had been marked as changed in ways that weren't obvious to me or the diff viewer (wasn't just whitespace, idk). Let me know if I accidentally discarded any real changes in the process.
- Breaks map rotation on aspect ratio change
- Upper/lower limits on scroll distance

**Review by Acumane — APPROVED** [review]
👌

File: src/components/home/MapItem.vue [hunk 1]
```diff
@@ -1,547 +1,605 @@
...
+      }
+    }
+  },
+  // General local variables
+  data() {
+    return {
+      threshold: 1,
+      doResize: "",
+      zoom: 0,
+      count:0,
```
Comment (Acumane) [inline src/components/home/MapItem.vue:387]: Unused

**Review by Acumane — APPROVED** [review]

**Pull request merged**